web/src/hash_util.ts

```diff
@@ -276,3 +276,16 @@ export function validate_group_settings_hash(hash: string): string {
     }
     return hash;
 }
+
+export function is_same_narrow(hash1: string, hash2: string): boolean {
+    if (hash1 === hash2) {
+        return true;
+    }
+
+    // Remove elements from hash that don't effect the narrow.
+    // TODO: Expand the regex to include more items we can ignore.
+    hash1 = hash1.replace(/near\/\d+$/, "");
+    hash2 = hash2.replace(/near\/\d+$/, "");
```

Comment:
This should be calculated using the `filter.ts` module helpers -- regexs will always be inaccurate, but we can check that they have the same set of sorted operators and operands, with perhaps the exception of a `near` operator in one or the other.